doc/release-notes/release-8.sgml.in

```diff
@@ -68,6 +68,13 @@ This section gives an account of those changes in three categories:
 
 </descrip>
 
+<sect1>Helper-related changes
```

Comment:
The proposed new `<sect1>Helper-related changes` subsection will "clash" with other `<sect1>` subsections in this `<sect>Changes to squid.conf` section when a single change affects a directive _and_ is related to a helper.

I suggest refactoring release notes structure, so that we can describe changes without having to force them into one of the narrow subsections _and_ add a simple _index_ of affected directive names, helper names, option names, etc. so that readers can easily/quickly check whether there is something of interest to them (and then use search to find related discussions in the same release notes document); the entries in that index will have no text/discussions.

This refactoring may also be a good time for switching to Markdown for (new) release notes.

Reply:
I am not keen to expand the scope to refactoring release notes, which is its own project.
The best way I can think about is to stack the three PRs @yadij is requesting on top of each other.
And since github doesn't support stacking, the best thing I can do is implement one after the other.

I'll need a week, probably. More if we allow the scope to drift. I'm obviously not happy but
there's nothing much else I can do

Reply:
> I am not keen to expand the scope to refactoring release notes, which is its own project.

Yes, that refactoring is its own project, requiring its own PR. IMO, it is up to _you_ whether to merge the (now three!) problematic PRs before or after those refactoring changes are going to address those problems. The tradeoffs seem clear to me, and I was OK with the (original, single) PR being merged already. Your call.

Reply:
These helper removals should be documented as a major feature change. see http://www.squid-cache.org/Versions/v4/squid-4.17-RELEASENOTES.html#ss2.4 for example.

There should also be entries for the changed `./configure` options, since they can no longer be passed these helper names. see the description of `--enable-auth-basic` in http://www.squid-cache.org/Versions/v4/squid-4.17-RELEASENOTES.html#ss4.2

Reply:
> These helper removals should be documented as a major feature change. see http://www.squid-cache.org/Versions/v4/squid-4.17-RELEASENOTES.html#ss2.4 for example.

Ok

> There should also be entries for the changed `./configure` options, since they can no longer be passed these helper names. see the description of `--enable-auth-basic` in http://www.squid-cache.org/Versions/v4/squid-4.17-RELEASENOTES.html#ss4.2

Are you sure? It's not an option that's being removed, it's an option *value*. With some specific exceptions, I do not think we document these, do we?

Reply:
Yes, Its a UI change to the values which are valid on that option.